setup.py

```diff
@@ -24,7 +24,7 @@ def package_version():
     url="https://github.com/noxdafox/pebble",
     packages=find_packages(exclude=["test"]),
     long_description=open(os.path.join(CWD, 'README.rst')).read(),
-    python_requires=">=3.6",
+    python_requires=">=3.9",
```

Comment:
As said, let's try to support `3.6` until possible. Type hinting is not a functional feature hence it does not justify dropping backwards compatibility.

Keep in mind that we use [`semver`](https://semver.org/) here. Hence, by doing the above, we should bump Pebble version to `6.0.0`. Not sure it's the right choice. :smile: